### PR TITLE
new cask: DevEco Studio

### PR DESCRIPTION
**Pull request by cyxmon**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by bevanjkay** [comment]
@cyxmon Can you please explain the issue you are trying to resolve here, as we may be able to help?

**Comment by cyxmon** [comment]
Sorry about the messy commit naming. Not quite much problems left now. If you would, could you tell me how to pass that commit style check about rebasing?

**Comment by bevanjkay** [comment]
@cyxmon We can fix the Commit style issue at the end. What is the reason for including a `preflight` block?

**Comment by cyxmon** [comment]
This cask is downloaded as a zip file, and the dmg file is in the directory of what's unarchived. Adding a preflight block is what I could think of to install the cask normally. Though, I think there should be some better way to do this but is currently beyond my knowledge.

**Comment by bevanjkay** [comment]
@cyxmon I'll have a look at fixing it now

**Comment by bevanjkay** [comment]
@cyxmon Have a look at the `container nested:` stanza, it should do what you are looking for - https://docs.brew.sh/Cask-Cookbook#optional-stanzas

**Comment by cyxmon** [comment]
Thanks, it works elegantly now.

File: Casks/d/deveco.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+cask "deveco" do
+  version "3.1.0.501"
+
+  on_arm do
+    sha256 "f3e77ba60e596c9e49cd5fc3ab67f3f944efd235f2fa7b298b501abcfc668f04"
+
+    url "https://contentcenter-vali-drcn.dbankcdn.cn/pvt_2/DeveloperAlliance_package_901_9/b7/v3/ONY_R_ihQ6-6k8Fnu8Kl2Q/devecostudio-mac-arm-#{version}.zip?HW-CC-KV=V1&HW-CC-Date=20230621T074841Z&HW-CC-Expire=315360000&HW-CC-Sign=FE223D9B6AB5C722096FF8C7BA574555E63638374A3D3543C72D7D9E869AB934",
```
Comment (bevanjkay) [inline Casks/d/deveco.rb:7]: @cyxmon It looks like there may be additional parts of this `url` that change between releases? I am also concerned about the use of query strings that may expire when time passes?

Comment (cyxmon) [reply Casks/d/deveco.rb:7]: You're right. This is what I am concerning as well. They just distribute releases in this way that makes it hard to follow up, which may require manual maintenance. How do you often deal with this situation? I'm not sure whether it's possible to find another stabler link.

Comment (bevanjkay) [reply Casks/d/deveco.rb:7]: @cyxmon If the `url` always requires a token that is refreshed then unfortunately this may not be able to be added to `homebrew-cask`. Acceptable casks should be easily-maintainable, and not require additional intervention on version updates than what can be achieved with the `brew bump` or `brew bump-cask-pr` commands.

Comment (cyxmon) [reply Casks/d/deveco.rb:7]: Make sense. I'll try to reach out to the official to see if they could provide a better solution.

**Comment by miccal** [comment]
I am going to merge this and we will see how stable the `url`'s are in time.

Thank you @cyxmon.

**Review by miccal — APPROVED** [review]
